src/TextField/TextField.js

```diff
@@ -98,7 +98,7 @@ export type Props = {
    *
    * @param {object} event The event source of the callback
    */
-  onChange?: (event: SyntheticInputEvent<>) => void,
+  onChange?: (event: SyntheticInputEvent<HTMLInputElement>) => void,
```

Comment:
This one should be `HTMLElement` as well, as the `InputProps`, which could change the base element type, are spread to `Input`

Reply:
Oh, the docs are misleading.
The docs for the `type` prop states the following:
* for `TextField`: 
```Type attribute of the `Input` element. It should be a valid HTML5 input type.```
* for `Input`:
```Type of the input element. It should be a valid HTML5 input type.```

I'll go ahead and change this to a `HTMLElement` as well. I'm not sure how you want to proceed w/ the docs